### PR TITLE
Catchup fixups, plugin URLs and VOD catchup mode

### PR DESCRIPTION
**Pull request by phunkyfish**
v5.2.0
- Added: Enable timeshift feature from inputstream.ffmpegdirect
- Added: Support VOD catchup streams including those via plugin URLs
- Added: Access to ffmpegdirect settings and inputstream checks (C++ API)
- Fixed: Fix flussonic ts detected as wrong type
- Fixed: Use kodis built in inputstreams for plugin urls
- Fixed: Fix incorrect catchup granularity for flussonic stream

### CODE REVIEW

File: src/iptvsimple/data/Channel.cpp [hunk 1]
```diff
@@ -198,10 +200,11 @@ void Channel::TryToAddPropertyAsHeader(const std::string& propertyName, const st
 
 void Channel::SetCatchupDays(int catchupDays)
 {
-  if (catchupDays != 0)
+  if (catchupDays > 0)
```
Comment (michaelarnauts) [inline src/iptvsimple/data/Channel.cpp]: When `catchupDays` == -1 (ignore), this doesn't do anything.

Comment (phunkyfish) [reply src/iptvsimple/data/Channel.cpp]: Faceplam

Comment (michaelarnauts) [reply src/iptvsimple/data/Channel.cpp]: Hmm, this might be okay, but the default is set to 0 in Channel::Reset

Comment (phunkyfish) [reply src/iptvsimple/data/Channel.cpp]: The default is fine. We will always set it for VOD type.

**Pull request merged**